### PR TITLE
Improve "authenticating to k8s api from a pod" docs

### PR DESCRIPTION
**Pull request by ahmetb**
Helps with https://github.com/kubernetes/kubernetes/issues/22121.
Also removing the link to kubectl-container example which is staged for removal.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6259)

### CODE REVIEW

**Comment by ahmetb** [comment]
@clhodapp I have a discussion about it at https://groups.google.com/d/msg/kubernetes-sig-cli/yUkt_ZkAAY0/WQ3NVRLyCwAJ feel free to participate if you support the idea we should have a kubectl-only image.

**Comment by chenopis** [comment]
@ahmetb FYI, the Travis CI error has been fixed. Please rebase or update the branch. Thx!

**Comment by ahmetb** [comment]
Rebased.

File: docs/tasks/administer-cluster/access-cluster-api.md [hunk 1]
```diff
@@ -188,17 +193,20 @@ used to verify the serving certificate of the apiserver.
...
 
-  - run a kubectl proxy as one of the containers in the pod, or as a background
-    process within a container.  This proxies the
-    Kubernetes API to the localhost interface of the pod, so that other processes
-    in any container of the pod can access it.  See this [example of using kubectl proxy
-    in a pod](https://github.com/kubernetes/examples/tree/{{page.githubbranch}}/staging/kubectl-container/).
-  - use the Go client library, and create a client using the `rest.InClusterConfig()` and `kubernetes.NewForConfig()` functions.
-    They handle locating and authenticating to the apiserver. [example](https://git.k8s.io/client-go/examples/in-cluster-client-configuration/main.go)
+  - if you would like to query the API without an official client library, you can run `kubectl proxy`
+    as one of the containers in the pod (i.e. sidecar container), or as a background process within a container.
```
Comment (tengqm) [inline docs/tasks/administer-cluster/access-cluster-api.md]: >as one

in one

Comment (tengqm) [reply docs/tasks/administer-cluster/access-cluster-api.md]: or:

```
 as the [command](/docs/tasks/inject-data-application/define-command-argument-container/) for a container
``

Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: actually rephrasing this to remove "background process" recommendation and just saying sidecar proxy with `kubectl proxy` as the command.

File: docs/tasks/administer-cluster/access-cluster-api.md [hunk 2]
```diff
@@ -168,17 +168,22 @@ There are [client libraries](/docs/reference/client-libraries/) for accessing th
...
-When accessing the API from a pod, locating and authenticating
+When accessing the API from a Pod, locating and authenticating
 to the API server are somewhat different.
 
-The recommended way to locate the apiserver within the pod is with
-the `kubernetes` DNS name, which resolves to a Service IP which in turn
-will be routed to an apiserver.
+The easiest way to use the Kubernetes API from a Pod is to use
+one of the official [client libraries](/docs/reference/client-libraries/). These
+libraries can automatically discover the apiserver and authenticate automatically.
```
Comment (steveperry-53) [inline docs/tasks/administer-cluster/access-cluster-api.md]: I don't think you need the second "automatically".
... can automatically discover the API server and authenticate.

File: docs/tasks/administer-cluster/access-cluster-api.md [hunk 3]
```diff
@@ -168,17 +168,22 @@ There are [client libraries](/docs/reference/client-libraries/) for accessing th
...
 
-The recommended way to authenticate to the apiserver is with a
-[service account](/docs/user-guide/service-accounts) credential.  By kube-system, a pod
+While running in a Pod, the Kubernetes apiserver is accessible via a Service named
+`kubernetes` in the `default` namespace. Therefore, Pods can use the
```
Comment (steveperry-53) [inline docs/tasks/administer-cluster/access-cluster-api.md]: the apiserver -> the API server

File: docs/tasks/administer-cluster/access-cluster-api.md [hunk 4]
```diff
@@ -188,17 +193,21 @@ used to verify the serving certificate of the apiserver.
...
-    Kubernetes API to the localhost interface of the pod, so that other processes
-    in any container of the pod can access it.  See this [example of using kubectl proxy
-    in a pod](https://github.com/kubernetes/examples/tree/{{page.githubbranch}}/staging/kubectl-container/).
-  - use the Go client library, and create a client using the `rest.InClusterConfig()` and `kubernetes.NewForConfig()` functions.
-    They handle locating and authenticating to the apiserver. [example](https://git.k8s.io/client-go/examples/in-cluster-client-configuration/main.go)
+  - if you would like to query the API without an official client library, you can run `kubectl proxy`
+    as the [command](/docs/tasks/inject-data-application/define-command-argument-container/)
+    of a new sidecar container in the Pod. This way, `kubectl proxy` will authenticate
+    to the API and expose it on the `localhost` interface of the Pod, so that other containers
+    in the Pod can use it directly.
```
Comment (steveperry-53) [inline docs/tasks/administer-cluster/access-cluster-api.md]: Start bullet items with caps.

- Use one ...
- If you would like ...

File: docs/tasks/administer-cluster/access-cluster-api.md [hunk 5]
```diff
@@ -188,17 +193,21 @@ used to verify the serving certificate of the apiserver.
 Finally, the default namespace to be used for namespaced API operations is placed in a file
 at `/var/run/secrets/kubernetes.io/serviceaccount/namespace` in each container.
 
-From within a pod the recommended ways to connect to API are:
+From within a Pod, the recommended ways to connect to the Kubernetes API are:
+
+  - use one of the official [client libraries](/docs/reference/client-libraries/)
+    as they handle API host discovery and authentication out of the box.
+    For Go client, the `rest.InClusterConfig()` function assists with this.
```
Comment (steveperry-53) [inline docs/tasks/administer-cluster/access-cluster-api.md]: out of the box -> automatically

File: docs/tasks/administer-cluster/access-cluster-api.md (hunk 3 above)
Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: We currently don't have guidance on what it should be, hence my confusion. 
https://www.google.com/search?q=site%3Akubernetes.io+"apiserver"&oq=site%3Akubernetes.io+"apiserver"
`apiserver` is indeed a Kubernetes term and it refers to the `kube-apiserver` component, but in this case I think we're trying to say "the API server"

Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: done

File: docs/tasks/administer-cluster/access-cluster-api.md (hunk 4 above)
Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: done

File: docs/tasks/administer-cluster/access-cluster-api.md (hunk 5 above)
Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: done

File: docs/tasks/administer-cluster/access-cluster-api.md (hunk 2 above)
Comment (ahmetb) [reply docs/tasks/administer-cluster/access-cluster-api.md]: done

**Comment by steveperry-53** [comment]
@ahmetb Are you ready for me to merge this?

**Pull request merged**